### PR TITLE
Fix witness generation and HoareMap

### LINKED ISSUE #195: Fix `HoareMap` and witness generation after `MapDomain` changes
In #164 the signatures for `MapDomain` were made stricter. Previously `HoareMap` abused some of the internal access, which was thus changed to at least compile, but untested and likely broke the already weird `HoareMap` and by extension witness generation.

### TODO
- [x] Fix `HoareMap` to make its domain tests pass again.
- [x] Confirm that witness generation works as before.

### PR DESCRIPTION
**Pull request by sim642**
Closes #195.

1. Remove `globally_unique_vids` which breaks original variable name mapping for witness invariants. See: https://github.com/goblint/cil/issues/31. AFAIK this isn't necessary because CIL already generates unique `vid`s. They just won't be consecutive after merging files, but that's not really important.
2. Use a Hoare domain without top for `PathSensitive2`. `SetDomain.Hoare` is based on `SetDomain.ToppedSet`, but the top isn't really meaningful for the path set. AFAIK there's no way for it to arise since it would mean an infinite number of paths. Removing the unnecessary lifting simplifies the logic.
3. Make `HoareMap` also top-less for `PathSensitive3`. This more or less matches the changes done by #164, because the lifted map domain doesn't expose operations on the inner map domain. The logic simplifies here even more.
4. Re-enable domain tests for `HoareMap`. An additional exclusion is necessary to not test `top`-related properties, since they're no longer valid.

### CODE REVIEW

**Comment by sim642** [comment]
@michael-schwarz Is there anything you still want to review about this or can I just merge?

**Comment by michael-schwarz** [comment]
@sim642: I haven't gotten around to looking at it yet, but will do so right now!

File: src/domains/domainProperties.ml [hunk 1]
```diff
@@ -117,7 +117,14 @@ struct
   include DomainTest (D)
 
   (* TODO: remove this hack *)
-  let has_top = try ignore (D.top ()); true with Failure _ -> false
+  let has_top =
+    try
+      ignore (D.top ());
+      true
+    with
+      | Failure _ (* raised by IntDomain *)
```
[the remark this answers is not in the document]
Comment (michael-schwarz) [reply src/domains/domainProperties.ml:125]: Actually, nevermind, it seems to be a lot of places where we do this, so it's probably not worth changing.

**Comment by michael-schwarz** [comment]
Ok, lgtm!

File: src/domains/domainProperties.ml (hunk 1 above)
[the remark this answers is not in the document]
Comment (sim642) [reply src/domains/domainProperties.ml:125]: I thought about it too, but both were widely used so it wasn't worth changing indeed. Really this should at some point be fixed in `DomainProperties` to allow testing domains without top explicitly.

**Pull request merged**